### PR TITLE
Add mod count to mods page

### LINKED ISSUE #451: Show mod count in mods tab
### Role

Improve modded minecraft experience

### Suggestion

Show mod count near the "help" button

### Benefit

Just because why not

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

Here's a lazy mockup:
image

### PR DESCRIPTION
**Pull request by Trial97**
Fixes #451

### CODE REVIEW

File: launcher/ui/pages/instance/ExternalResourcesPage.cpp [hunk 1]
```diff
@@ -248,6 +256,15 @@ bool ExternalResourcesPage::onSelectionChanged(const QModelIndex& current, const
...
-
     return true;
 }
+
+QString ExternalResourcesPage::extraHeaderInfoString()
+{
+    if (ui && ui->treeView && ui->treeView->selectionModel()) {
+        auto selection = m_filterModel->mapSelectionToSource(ui->treeView->selectionModel()->selection()).indexes();
+        if (auto count = std::count_if(selection.cbegin(), selection.cend(), [](auto v) { return v.column() == 0; }); count != 0)
+            return tr("[%1 installed, %2 selected]").arg(m_model->size()).arg(count);
```
Comment (getchoo) [inline launcher/ui/pages/instance/ExternalResourcesPage.cpp]:
```suggestion
            return tr(" [%1 installed, %2 selected]").arg(m_model->size()).arg(count);
```
this looks a little cramped imo
image

**Review by getchoo — CHANGES_REQUESTED** [review]
i'm not exactly sure what's going on here: in my testing the mods page (and only the mods page) will have the install/select count appear when you first view it, but if you switch to a different tab like resource packs and go back, it won't appear at all until you have selected a mod.

File: launcher/ui/pages/instance/ExternalResourcesPage.cpp (hunk 1 above)
Comment (TheKodeToad) [reply launcher/ui/pages/instance/ExternalResourcesPage.cpp]: I think rounded brackets would be better

**Comment by Trial97** [comment]
I fixed the mentioned comment but in order to fix all of them I removed a line that was prefixed by the following comment: 
```
// Prevent a weird crash when trying to open the mods page twice in a session o.O
disconnect(mods.get(), &ModFolderModel::updateFinished, this, 0);
```

I could not get it to crash as the description is vague. But if somebody knows why that line was added let me know.

**Comment by getchoo** [comment]
it seems @flowln was the one who left this (sorry for the ping, not sure who else to ask :p). could you shed some light on this so we can see if it's still needed?

**Comment by flowln** [comment]
The description isn't vague, it is just as simple as it sounds: Without that line, it used to be the case that the application would crash if you opened the mods page, switched pages, and returned back to the mods page. I suppose the underlying cause of the crash got fixed at some point, so the line doesn't do much now. Nice!

Also, you can probably remove `mods` from the lambda now.

**Comment by getchoo** [comment]
thank you so much! just wanted to make sure :)

**Review by getchoo — APPROVED** [review]
i just tested this again as well and it seems to good to go! the only other thing i would recommend is following flow's suggestion

**Review by DioEgizio — APPROVED** [review]
seems to work as expected

**Comment by getchoo** [comment]
thanks!

**Pull request merged**